homeassistant/components/meteoswiss/weather.py

```diff
@@ -0,0 +1,122 @@
+"""Support for Met.no weather service."""
+
+import datetime
+
+import meteoswiss_async
+
+from homeassistant.components.weather import (
+    Forecast,
+    SingleCoordinatorWeatherEntity,
+    WeatherEntityFeature,
+)
+from homeassistant.const import (
+    UnitOfPrecipitationDepth,
+    UnitOfPressure,
+    UnitOfSpeed,
+    UnitOfTemperature,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import MeteoSwissConfigEntry
+from .const import DOMAIN, LOCATION_NAME
+from .coordinator import MeteoSwissDataUpdateCoordinator
+
+DEFAULT_NAME = "MeteoSwiss"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: MeteoSwissConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add a weather entity from a config_entry."""
+    coordinator = config_entry.runtime_data
+
+    entities = [MetWeather(coordinator, config_entry)]
+
+    async_add_entities(entities)
+
+
+class MetWeather(SingleCoordinatorWeatherEntity[MeteoSwissDataUpdateCoordinator]):
+    """Implementation of a MeteoSwiss weather condition."""
+
+    _attr_attribution = (
+        "Weather forecast from MeteoSwiss, delivered by the Swiss "
+        "Meteorological Institute."
+    )
+    _attr_has_entity_name = True
+    _attr_native_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_native_precipitation_unit = UnitOfPrecipitationDepth.MILLIMETERS
+    _attr_native_pressure_unit = UnitOfPressure.HPA
+    _attr_native_wind_speed_unit = UnitOfSpeed.KILOMETERS_PER_HOUR
+    _attr_supported_features = (
+        WeatherEntityFeature.FORECAST_DAILY | WeatherEntityFeature.FORECAST_HOURLY
+    )
+
+    def __init__(
+        self,
+        coordinator: MeteoSwissDataUpdateCoordinator,
+        config_entry: MeteoSwissConfigEntry,
```

Comment:
I personally prefer to not inject the config entry because it always feels like a very clunky object to inject